orttraining/orttraining/core/framework/gradient_graph_builder.h

```diff
@@ -88,6 +88,10 @@ class GradientGraphBuilder {
 
   Status Build(const std::unordered_set<std::string>* p_initializer_names_to_preserve = nullptr);
 
+  std::unordered_set<std::string> GetNonDifferentiableYNodeArgNames() const {
```

Comment:
Performance nit: return `const std::unordered_set<std::string>&` (and change call-site to assign the return value to a reference)?

Reply:
done.